src/components/PopupWithConfirmation.js

```diff
@@ -0,0 +1,25 @@
+import Popup from "./Popup.js";
+
+export default class PopupWithConfirmation extends Popup {
+  constructor(popupSelector) {
+    super({ popupSelector });
+    this._yesDeleteButton = document.querySelector("#delete-card-button");
+    this._saveButton = this._popupElement.querySelector(".modal__save");
+  }
+
+setSubmitAction(action) {
+    this._handleSubmit = action;
+}
+openPopup() {
+    this._saveButton.textContent = "Yes";
```

Comment:
You need to return the **default** button text only in blocks `finally` so that the button could get the text back in any case: `success` or `failure`.